src/PhpBrew/VariantBuilder.php

```diff
@@ -551,6 +552,26 @@ public function __construct()
             }
         };
 
+        $this->variants['sodium'] = function (Build $build, $val = null) {
+            if ($build->compareVersion('7.2') < 0) {
```

Comment:
Shouldn't the user just not include the variant that is not supported by a given PHP version? Otherwise, if a process of building PHP is automated, the user would expect to get the extension available upon successful completion but it won't be true if we just omit this variant for PHP 7.1 and older.

On the other hand, if sodium is unpacked under the `ext` directory and `phpize`d, it probably may work even with older PHPs, so PHPBrew shouldn't prevent that.

Reply:
In PHP 7.1 and older, sodium isn't bundled with PHP and `--with-sodium` isn't recognized by configure. If you use `--with-sodium` with older PHP versions the build will just succeed, but you won't have sodium. Removing this check doesn't solve the issue of automated builds you're describing.

Without the warning, the only indication that something went wrong is a line the build log

    configure: WARNING: unrecognized options: --with-sodium

This typically goes unnoticed.

Reply:
Manually adding extensions to the build dir and using variants to include them isn't supported by phpbrew. Variants can only be used for extensions that are bundled. I don't see why this should be the exception.

Reply:
Then this is just a replacement of one more or less standard warning with another warning specific to PHPBrew. It's better if we keep the standard one which is easier to google and understand what's going on.

Reply:
This warning is given on-screen, not just in the log. It's similar to warnings for other extensions like

    echo "homebrew prefix '$output' doesn't exist. you forgot to install?\n";

I did just remove the `return`. In other cases, the option is added regardless of the warning. Here it's now the same.

Reply:
Yeah, that should be fine.